### ISSUE TITLE
Bug: Unit Test fails to run on focused project

### ISSUE THREAD

**Issue by yongjincho92** [MobileNativeFoundation/rules_xcodeproj#2166]
### Description

Unit test execution fails due to missing xcframework in Xcode project generated with unit test target as a focused target 

### Reproduction steps

Repro steps are provided in the following change : https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/2165

### Expected behavior

Tests should succeed 

### rules_xcodeproj version

7efb52807d2ff37be1ed161f992c0ec1c635b1d0

### Xcode version

14.1

### Bazel version

6.1

### rules_apple version

n/a

### rules_swift version

n/a

**Comment by brentleyjones** [comment]
Yeah, similar to how in BwX mode we require various dependencies to stay focused, we probably need to unconditionally keep the test host of a target focused as well.
